vault/eventbus/bus.go

```diff
@@ -115,8 +153,8 @@ func (bus *EventBus) Subscribe(_ context.Context, eventType logical.EventType) (
 		return nil, err
 	}
 
-	// TODO: should we have just one node, and handle all the routing ourselves?
-	asyncNode := newAsyncNode()
+	// TODO: should we have just one node per namespace, and handle all the routing ourselves?
+	asyncNode := newAsyncNode(ns)
```

Comment:
I think we should probably error if a `nil` namespace is passed in. We have `namespace.RootNamespace` for the base namespace, so there's no reason for it to be omitted.

Reply:
Agreed.  Though there's a convention (which I don't love) in Vault of passing namespace arguments via the context arg.  Not asking for you to adopt that here, just want to make sure you've considered it.

Reply:
Yep. Initially I did that, but Tom suggested making it explicit here instead (which I kind of prefer now).